src/coreclr/jit/hwintrinsiccodegenarm64.cpp

```diff
@@ -394,8 +394,6 @@ void CodeGen::genHWIntrinsic(GenTreeHWIntrinsic* node)
                     }
                     else if (isRMW)
                     {
-                        assert(targetReg != op2Reg);
```

Comment:
This assert is required because the instruction is `rmw`, that is it both reads and modifies the destination register.

Basically, we have an operation `targetReg = Ins(op1Reg, op2Reg)`, because its `RMW` the operation is actually: `Ins(ref op1Reg, op2Reg)`.
This means that for the result to actually end up in `targetReg` we have to emit this as:
```
targetReg = op1Reg;
Ins(ref targetReg, op2Reg);
```

If `targetReg == op2Reg`, then what we'll end up emitting is equivalent to:
```
op2Reg = op1Reg;
Ins(ref op2Reg, op2Reg);
```

Thus it will no longer do the right thing and the code is invalid.